### PR TITLE
M3-2465 Update Auth Code

### PR DESCRIPTION
**Pull request by martinmckenna**
## Description

Make the auth code look like all our other redux code

### Todo
- [x] Make AuthenticationWrapper.test.tsx work
- [x] Make OAuth.test.tsx work
- [ ] Make an API request to invalidate token upon logout (this will be handled in another PR)

### To Test
1. Make sure that local storage and redux state is being set upon logging in
2. Clear local storage and refresh the page - the app should redirect you to login, then re-set local storage and redux state
3. Using a proxy like Charles, test 401 responses - the app should revalidate you and update local storage and redux state
4. Make sure there are no cases where the user can navigate the app without first being authed

### CODE REVIEW

**Comment by johnwcallahan** [comment]
Nice fix on the issue above.

There could be a flaw in the way I'm testing this, but here's what I'm running into:

1) Login to the app. 
2) Open local storage and set `authentication/expire` to a date in the past.
3) Refresh the page.
4) Observe: blank screen. Local storage has been cleared, but we are not redirected to login.

I think we'd need to redirect to login in [this](https://github.com/linode/manager/pull/4564/files#diff-84ee98a872175070fe999f3cab06f23fR49) case, but there might be a better place to do it.

File: src/store/authentication/authentication.reducer.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      expiration: expires || null
+    };
+  })
+  .case(handleInitTokens, state => {
+    /**
+     * if our token is expired, clear local storage
+     * and redux state
+     */
+    const expiryTime = expiryInLocalStorage.get();
+    if (expiryTime && new Date(expiryTime) < new Date()) {
```
Comment (asauber) [inline src/store/authentication/authentication.reducer.ts]: This is why we multiply the expiration timestamp by 1000, the API returns epoch time in seconds, JavaScript uses millisecond epoch time.

File: src/App.tsx [hunk 2]
```diff
@@ -271,12 +271,47 @@ export class App extends React.Component<CombinedProps, State> {
...
     if (hasError) {
       return <TheApplicationIsOnFire />;
     }
 
+    /**
+     * basically, if we get an "invalid oauth token"
+     * error from the API, just render nothing because the user is
+     * about to get shot off to login
+     */
+    if (
```
Comment (asauber) [inline src/App.tsx]: I would highly recommend doing this in a generic way, so that the functionality does not depend on maintaining this list.

Comment (martinmckenna) [reply src/App.tsx]: agreed. We have a spike to re-work the global errors anyway, so this will be an item to address on that ticket

**Pull request merged**